### PR TITLE
Add L2 solver for FUGW

### PR DESCRIPTION
**Pull request by 6Ulm**
Add solver when all the divergences in FUGW are L2-squared (instead of KL)

### CODE REVIEW

**Comment by alexisthual** [comment]
Tests seem to fail because if timeouts, however your last change seems minimal so I don't know what's going on.
Maybe you could push an empty commit to restart the tests.
If they fail again, could you make sure they run fast on a non-GPU setup?

File: src/fugw/solvers/dense.py [hunk 1]
```diff
@@ -20,7 +21,7 @@ class FUGWSolver(BaseSolver):
     """Solver computing dense solutions"""
 
     def local_biconvex_cost(
-        self, pi, transpose, data_const, tuple_weights, hyperparams
+        self, pi, transpose, data_const, tuple_weights, hyperparams, divergence
```
Comment (alexisthual) [inline src/fugw/solvers/dense.py:24]: We should probably try to fit `divergence` in `hyperparams`, because all current variables which define the shape of loss function are in `hyperparams` :slightly_smiling_face:

File: src/fugw/solvers/dense.py [hunk 2]
```diff
@@ -57,27 +58,30 @@ def local_biconvex_cost(
...
-            cost += eps * entropic_cost
+            if reg_mode == "joint":
+                entropic_cost = compute_approx_kl(pi, ws_dot_wt)
+                cost += eps * entropic_cost
 
         return cost
 
-    def fugw_loss(self, pi, gamma, data_const, tuple_weights, hyperparams):
+    def fugw_loss(
+        self, pi, gamma, data_const, tuple_weights, hyperparams, divergence
```
Comment (alexisthual) [inline src/fugw/solvers/dense.py:77]: Same here

**Review by alexisthual — COMMENTED** [review]
LGTM but still have a minor comment :)

File: src/fugw/solvers/dense.py (hunk 1 above)
Comment (6Ulm) [reply src/fugw/solvers/dense.py:24]: I agree, I will put all in `hyperparams`

**Review by alexisthual — APPROVED** [review]
LGTM!

**Comment by alexisthual** [comment]
Merging!
